tools/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,582 @@
+        3. Once all jobs have been recorded, complete.
+        """
+        start_time = time.time()
+
+        # Wait for the training config to be available. This indicates the training has begun.
+        while True:
+            if time.time() - start_time > _TRAINING_CONFIG_WAIT_SECONDS:
+                raise RuntimeError(
+                    f"Training config still doesn't exist after:"
+                    f"{_TRAINING_CONFIG_WAIT_SECONDS / 60} minutes"
+                )
+
+            if PathManager.exists(self.training_checkpoint_file):
+                # Load training yaml config.
+                self._load_training_config()
+
+                # Build main training task in order to extract iteration info.
+                self.training_task = build_task(self.training_config)
+                self.training_task.dataloaders = self.training_task.build_dataloaders(
+                    False
+                )
+
+                # Generate evaluation results
+                self.evaluation_results = self._generate_initial_benchmark_results()
+                self._validate_evaluation_setup()
+
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+        # Save initial evaluation benchmarks, for checkpointing reasons.
+        self.save_evaluation_benchmarks()
+
+        # Checkpoint folder is now available. Continuously monitor the training checkpoints,
+        # launch evaluation jobs as needed, monitor their progress, and record their results.
+        while True:
+            self._evaluate_checkpoints()
+
+            self._check_evaluation_jobs()
+
+            # Break if no more checkpoints to evaluate
+            if self._finished():
+                logging.info("Evaluations are finished")
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+    def save_evaluation_benchmarks(self):
+        """
+        Create the /evaluations directory inside the training checkpoints dir.
+        Upload json file to the parent evaluation directories, as well as
+        to each child evaluation directories.
+        """
+        # Upload all checkpoints evaluations to parent checkpoint directory.
+        evaluation_dir = self.evaluation_dir()
+        parent_metrics_file = os.path.join(evaluation_dir, "evaluation_metrics.json")
+
+        if not PathManager.exists(evaluation_dir):
+            PathManager.mkdirs(evaluation_dir)
+
+        self._write_json_file(self.evaluation_results, parent_metrics_file)
+
+        # Upload each checkpoint's evaluations to child directories.
+        for checkpoint_str, benchmarks in self.evaluation_results.items():
+            child_metrics_dir = os.path.join(evaluation_dir, checkpoint_str)
+            child_metrics_file = os.path.join(
+                child_metrics_dir, "evaluation_metrics.json"
+            )
+
+            if not PathManager.exists(child_metrics_dir):
+                PathManager.mkdirs(child_metrics_dir)
```

Comment:
nit: can we use this function directly: https://github.com/facebookresearch/vissl/blob/master/vissl/utils/io.py#L122-L133 ?